# Практичне завданняя по масивам `NumPy`

Намагайтесь виконати завдання, не викоритовуючи циклів. Більшість завдань потребують 1-2 строки коду

0. Підготовка вхідних даних.

Переконайтесь, що модуль `NumPy` встановлено на ваш комп'ютер.
Виконайте наступний фрагмент коду, не вносячи ніяких змін. Він створить масив `data` з даними до завдання

In [1]:
# не змінюйте код у цій ячейці

import numpy as np
from numpy import random

np.random.seed(12345) # случайность неслучайна
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
print(data)


[4410  522   72 2700  738  170 1740   37  531  910 4186 2000 3139  704
 5929   80  648 1476  570  630 1376 2208 2494 3864 4565 2449 5978 2716
  637  348 1404  192  258  160  195 6300  910 2025  220  170  106 5976
  440 4704  205 2065  415  612  336 6240 2241 2318 3696 2304  470  888
  836 1210 1068 1550 3936 1746 4600 5084 3990 2714 5568 1771  777  161
  972 5166    0 8370   44 1425  896 1534 1118 1680 1188 2880   69 3840
  570 5082 3381 1776 4608 3182 1922 5676  882 4030 1824  245 3162 5040
 1065 6762]


1. Знайдіть найменше і найбільше значення в масиві `data`. Має получитись

0       
8370

In [6]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
min_number = min(data)
max_number = max(data)
print(min_number)
print(max_number)

0
8370


2. Визначте індекси, за якими розташовано найменше і найбільше значення в масиві `data`. Має получитись

72      
73

In [15]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
min_index=np.argmin(data)
max_index=np.argmax(data)
print(min_index)
print(max_index)

72
73


3. Перетворить масив `data` на матрицю розміром 10х10

In [21]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
rows, cols = 10, 10
data_matrix = data.reshape(rows, cols)
print(data_matrix)

[[4410  522   72 2700  738  170 1740   37  531  910]
 [4186 2000 3139  704 5929   80  648 1476  570  630]
 [1376 2208 2494 3864 4565 2449 5978 2716  637  348]
 [1404  192  258  160  195 6300  910 2025  220  170]
 [ 106 5976  440 4704  205 2065  415  612  336 6240]
 [2241 2318 3696 2304  470  888  836 1210 1068 1550]
 [3936 1746 4600 5084 3990 2714 5568 1771  777  161]
 [ 972 5166    0 8370   44 1425  896 1534 1118 1680]
 [1188 2880   69 3840  570 5082 3381 1776 4608 3182]
 [1922 5676  882 4030 1824  245 3162 5040 1065 6762]]


4. Знайдіть мінімальне значення в кожному рядку масиву `data`. Має получитись

array([ 37,  80, 348, 160, 106, 470, 161,   0,  69, 245])

In [20]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
rows, cols = 10, 10
data_matrix = data.reshape(rows, cols)
min_values_in_rows = [min(rows) for rows in data_matrix]
print(min_values_in_rows)


[37, 80, 348, 160, 106, 470, 161, 0, 69, 245]


5. Повертаємось до одномірного масиву `data`.


Знайдіть елементи масиву `data`, більші за 6000. Має бути

array([6300, 6240, 8370, 6762])

In [26]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
more_than_six_thousand = np.logical_not(data < 6000)
print(data[more_than_six_thousand])

[6300 6240 8370 6762]


6. Скільки в масиві `data` елементів, які більше за 500 та менші за 1000?

18

In [25]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
bigger_than_five_hundred_smaller_than_thousand = np.logical_and(data > 500, data <1000)
new_array = data[bigger_than_five_hundred_smaller_than_thousand]
count = len(new_array)
print(count)

18


7. Визначте суму чисел в масиві `data`.

216077

In [27]:
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
sum_list = sum(data)
print(sum_list)

216077


8. Замініть на `0` значення всіх єлементів, які менші ніж 100

In [31]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
data[data < 100] = 0
print(data)

[4410  522    0 2700  738  170 1740    0  531  910 4186 2000 3139  704
 5929    0  648 1476  570  630 1376 2208 2494 3864 4565 2449 5978 2716
  637  348 1404  192  258  160  195 6300  910 2025  220  170  106 5976
  440 4704  205 2065  415  612  336 6240 2241 2318 3696 2304  470  888
  836 1210 1068 1550 3936 1746 4600 5084 3990 2714 5568 1771  777  161
  972 5166    0 8370    0 1425  896 1534 1118 1680 1188 2880    0 3840
  570 5082 3381 1776 4608 3182 1922 5676  882 4030 1824  245 3162 5040
 1065 6762]


9. На скільки зменшилась сума елементів масиву `data` після виконання попереднього завдання?

302

In [35]:
# здесь може бути ваш код
np.random.seed(12345)
x=random.randint(100, size=(100))
y=random.randint(100, size=(100))
data=x*y
sum_list = sum(data)
data[data < 100] = 0
new_sum_list = sum(data)
print(sum_list - new_sum_list)

302
